packages/generator/src/edmx-parser/common/edmx-parser.ts

```diff
@@ -6,23 +7,52 @@ import {
 } from './edmx-types';
 
 export function parseComplexTypesBase(root): EdmxComplexTypeBase[] {
-  return forceArray(root.ComplexType).map(c => ({
+  return getMergedPropertyWithNamespace(root, 'ComplexType').map(c => ({
     ...c,
-    Property: forceArray(c.Property)
+    Property: forceArray(c.Property),
+    Namespace: c.Namespace
   }));
 }
 
 export function parseEntityTypesBase(root): EdmxEntityTypeBase<any>[] {
-  return forceArray(root.EntityType).map(e => ({
+  return getMergedPropertyWithNamespace(root, 'EntityType').map(e => ({
     ...e,
     Key: {
       PropertyRef: forceArray(e.Key?.PropertyRef)
     },
     NavigationProperty: forceArray(e.NavigationProperty),
-    Property: forceArray(e.Property)
+    Property: forceArray(e.Property),
+    Namespace: e.Namespace
   }));
 }
 
 export function parseEntitySetsBase(root): EdmxEntitySetBase[] {
-  return forceArray(root.EntityContainer.EntitySet);
+  return getPropertyFromEntityContainer(root, 'EntitySet');
+}
+
+export function getPropertyFromEntityContainer(
+  schema,
+  entityContainerProperty: string
+): any[] {
+  return flat(
+    forceArray(schema)
+      .filter(s => s.EntityContainer)
+      .map(s =>
+        forceArray(s['EntityContainer'][entityContainerProperty]).map(
+          addNamespace(schema.Namespace)
+        )
+      )
+  );
+}
+
+export function addNamespace(namespace) {
+  return obj => assoc('Namespace', namespace, obj);
+}
```

Comment:
Medium: Also consider, whether it would be nicer to not return a function here, but really only adding the namespace and using the callback in `.map`. I find this easier to read and then the name of the function fits what the function is actually doing.

```suggestion
export function addNamespace<T>(obj: T, namespace: string): T & { Namespace: string } {
  return { ...obj, Namespace: namespace };
}
```

Usage:
```ts
...map(obj => addNamespace(obj, schema.Namespace);
```